### PR TITLE
refactor: Migrate memcached to opendal based

### PR DESCRIPTION
**Pull request by Xuanwo**
This will migrate memcached to opendal based. Also, I added an integration tests too!

Close https://github.com/mozilla/sccache/pull/1539

### CODE REVIEW

File: src/cache/memcached.rs [hunk 1]
```diff
@@ -13,94 +13,25 @@
...
-
-    async fn current_size(&self) -> Result<Option<u64>> {
-        Ok(None)
-    }
-    async fn max_size(&self) -> Result<Option<u64>> {
-        Ok(None)
+    pub fn build(url: &str) -> Result<Operator> {
+        let mut builder = memcached::Builder::default();
+        builder.endpoint(url);
+        // Set default ttl to 24 hours.
```
Comment (sylvestre) [inline src/cache/memcached.rs:31]: could you please add the "why" 24 hours ?

**Comment by sylvestre** [comment]
Could you please rebase the cargo files?

**Comment by sylvestre** [comment]
it says now
"This branch cannot be rebased due to conflicts" ?!

**Comment by Xuanwo** [comment]
> it says now "This branch cannot be rebased due to conflicts" ?!

I have this PR squashed. Hoping this will address your problem :heart:

**Pull request merged**